File: java/src/test/java/problems1_100/SwapNodesInPairsTest.java
package problems1_100;

import concepts.SinglyListNode;
import org.junit.Test;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class SwapNodesInPairsTest extends SampledTest {
    @Test
    public void test() {
        for (SwapNodesInPairs solution : SwapNodesInPairs.values()) {
            for (SwapNodesInPairsSample sample : getSamples(SwapNodesInPairsSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.swapPairs(SinglyListNode.of(sample.input.head)))
                    .isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/SwapNodesInPairsTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class SwapNodesInPairsTest : SampledTest() {
    @Test
    fun test() =
        SwapNodesInPairs.entries.forEach { solution ->
            getSamples(Array<SwapNodesInPairsSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.swapPairs(SinglyListNode.of(*input.head)))
                        .isEqualTo(SinglyListNode.of(*output))
                }
        }
}
